src/smart-components/role/edit-resource-definitions-modal.js

```diff
@@ -0,0 +1,193 @@
+import React, { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import componentTypes from '@data-driven-forms/react-form-renderer/dist/cjs/component-types';
+import FormRenderer from '../common/form-renderer';
+import { useHistory, useRouteMatch } from 'react-router-dom';
+import { useDispatch, useSelector, shallowEqual } from 'react-redux';
+import { updateRole, fetchRole } from '../../redux/actions/role-actions';
+import { routes as paths } from '../../../package.json';
+import { getResource, getResourceDefinitions } from '../../redux/actions/cost-management-actions';
+import componentMapper from '@data-driven-forms/pf4-component-mapper/dist/cjs/component-mapper';
+import { WarningModal } from '../common/warningModal';
+import { Spinner, Modal } from '@patternfly/react-core';
+import ResourceDefinitionsFormTemplate from './ResourceDefinitionsFormTemplate';
+import './role-permissions.scss';
+
+const createOptions = (resources) => {
+  let options = [];
+  for (const [key, value] of Object.entries(resources)) {
+    options = [
+      ...options,
+      ...value.map((r) => ({
+        value: r.value,
+        path: key,
+        label: r.value,
+      })),
+    ];
+  }
+
+  return options;
+};
+
+const createEditResourceDefinitionsSchema = (resources, resourcesPath, options) => ({
+  fields: [
+    {
+      component: componentTypes.DUAL_LIST_SELECT,
+      name: 'dual-list-select',
+      leftTitle: 'Resources available for the permission',
+      rightTitle: 'Resources added to the permission',
+      filterOptionsTitle: 'Filter by resource...',
+      filterValueTitle: 'Filter by resource...',
+      options: [...(resourcesPath && resources ? options : [])],
+      validate: [{ type: 'validate-resources' }],
+    },
+  ],
+});
+
+const selector = ({ costReducer: { resourceTypes, isLoading, loadingResources, resources } }, resourcesPath) => ({
+  resourceTypes: resourceTypes.data,
+  resources: resources[resourcesPath] ? { resourcesPath: resources[resourcesPath] } : resources,
```

Comment:
I am pretty sure, that this piece of code is not doing what you want. You are creating object with `"resourcePath"` string as a key, that is not necessary. You can just select resources for current permission a be good with it. Also you should change `createOptions` function.
```
resources: resources[resourcesPath] || [],
```